### PR TITLE
Chain Evm Map

### PR DESCRIPTION
**Pull request by joeizang**

### CODE REVIEW

**Comment by ryestew** [comment]
I just tried switching in the Browser Wallet to Celo.  But the change in network in the wallet did not get reflected in Deploy & Run.

**Review by ryestew — CHANGES_REQUESTED** [review]

**Comment by ryestew** [comment]
Strangely when I tried 1st to deploy to Celo Alfajores, I got this out of gas error:
image

Then when I tried again I got the warning in this PR:
image

**Comment by joeizang** [comment]
> Strangely when I tried 1st to deploy to Celo Alfajores, I got this out of gas error: image
> 
> Then when I tried again I got the warning in this PR: image

I am not sure  about the gas error though.

File: libs/remix-ui/run-tab/src/lib/actions/evmmap.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+      { id: 369, name: "PulseChain" },
+      { id: 388, name: "HALO Mainnet" },
+      { id: 1088, name: "Metis Andromeda Mainnet" },
+      { id: 1101, name: "Polygon zkEVM" },
+      { id: 1284, name: "Moonbeam" },
+      { id: 2000, name: "Dogechain Mainnet" },
+      { id: 42220, name: "Celo Mainnet" },
+      { id: 42161, name: "Arbitrum One" },
+      { id: 42170, name: "Arbitrum Nova" },
+      { id: 44787, name: "Celo Alfajores Testnet" },
```
[the remark this answers is not in the document]
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/actions/evmmap.ts:106]: ok

File: libs/remix-ui/run-tab/src/lib/run-tab.tsx [hunk 2]
```diff
@@ -102,6 +106,67 @@ export function RunTabUI(props: RunTabProps) {
...
+        return 'Failed'
+      } else {
+        setEvmCheckComplete(true)
+        if (!IsCompatible) {
+        //show modal
+          plugin.call('notification', 'modal', {
+            id: 'evm-chainId-incompatible',
+            title: 'Incompatible EVM - ChainId Detected',
+            message: <div className="px-3">
+              <p>The selected chain is not compatible with the selected EVM version. Please select a one of the options below.</p>
```
[the remark this answers is not in the document]
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/run-tab.tsx:151]: Done

**Comment by ryestew** [comment]
I have one last question - if the EVM chain map gets outdated - can someone still deploy to the chain - they just hit the cancel button and then they deploy anyway - right?

**Comment by ryestew** [comment]
There is something weird with the modals - I tried to deploy to Celo mainnet and I got 2 modals - the first disappeared to fast to see it and the second was the old modal:
image

Something similar happens when I load this deployed instance - both entry modals appear and disappear.

File: libs/remix-ui/run-tab/src/lib/actions/evmmap.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joeizang) [reply libs/remix-ui/run-tab/src/lib/actions/evmmap.ts:108]: done

**Comment by ryestew** [comment]
The selected chain is not compatible with the EVM version that the Solidity contract had been compiled with. Please select a one of the options below.

If there is time, please change the above sentence to:
The smart contract has not been compiled with an EVM version that is compatible with the selected chain.

**Comment by joeizang** [comment]
> The smart contract has not been compiled with an EVM version that is compatible with the selected chain.

Done. Please check

**Review by ryestew — APPROVED** [review]

**Comment by LianaHus** [comment]
@joeizang we see a toaster about new settings which says "**udapp** is updating ..." Udapp is internal name and should be replaced by plugin _Display name_

**Comment by LianaHus** [comment]
The rest looks LGTM

**Pull request merged**